modules/ort/src/main/kotlin/workflow/processors/enricher/OrtDownloaderProcessor.kt

```diff
@@ -33,39 +34,45 @@ class OrtDownloaderProcessor : AbstractProcessor() {
+        sourcesZipDirectory = context.toolConfiguration.antennaTargetDirectory.resolve(ORT_DOWNLOADER_DIR).toFile()
+        if (!sourcesZipDirectory.isDirectory && !sourcesZipDirectory.mkdirs()) {
+            throw ConfigurationException("Failed to create directory '${sourcesZipDirectory.absolutePath}' " +
+                    "for ORT Downloader result. ")
+        }
     }
 
     override fun process(intermediates: MutableCollection<Artifact>): MutableCollection<Artifact> {
         intermediates.filterNot {
             it.sourceFile.isPresent
-        }.forEach {
-            val pkg = ArtifactToPackageMapper().apply(it)
-            val name = it.askFor(ArtifactCoordinates::class.java).takeIf { f -> f.isPresent }
-                    ?.let {
-                        it.get().mainCoordinate.canonicalize()
-                    } ?: it.askFor(ArtifactFilename::class.java).takeIf { f -> f.isPresent }
-                    ?.let { f -> f.get().bestFilenameEntryGuess.takeIf { fe -> fe.isPresent }
-                            ?.let { e -> e.get().filename }
-                    } ?: "unknown"
+        }.forEach { artifact ->
+            val pkg = ArtifactToPackageMapper().apply(artifact)
+            val name = artifact.askFor(ArtifactCoordinates::class.java).takeIf {
+                fact -> fact.isPresent
+            }?.get()?.mainCoordinate?.canonicalize() ?: artifact.askFor(ArtifactFilename::class.java).takeIf {
+                fact -> fact.isPresent
+            }?.let { filename ->
+                filename.get().bestFilenameEntryGuess.takeIf { filenameEntry ->
+                    filenameEntry.isPresent
+                }?.get()?.filename
+            } ?: "unknown"
             try {
                 LOGGER.debug("Download sources via ORT Downloader for '${name}'")
-                val downloadResult = Downloader().download(pkg, createTempDir("ortDownloaderDirectory"))
-                if (downloadResult.downloadDirectory.exists()) {
+                val ortDownloadDirectory = createTempDir("ortDownloaderDirectory").also { it.deleteOnExit() }
+                val downloadResult = Downloader().download(pkg, ortDownloadDirectory)
+                if (downloadResult.downloadDirectory.exists() && downloadResult.downloadDirectory.isDirectory) {
                     val zipFile = File(
-                            ortDownloadFolder,
+                            sourcesZipDirectory,
                             "${pkg.id.name.encodeOrUnknown()}-${pkg.id.version.encodeOrUnknown()}.zip"
                     )
                     LOGGER.debug("Pack source directory '${downloadResult.downloadDirectory.absolutePath}' to " +
                             "'${zipFile.absolutePath}.")
                     downloadResult.downloadDirectory.packZip(zipFile)
 
-                    if (zipFile.exists()) {
-                        it.addFact(ArtifactSourceFile(zipFile.toPath()))
+                    if (zipFile.exists() && zipFile.isFile) {
```

Comment:
`isFile` (and also `isDirectory`, BTW) imply `exists()`, so only either one is needed.